EngineLayer/PeptideSpectralMatch.cs

```diff
@@ -268,6 +277,117 @@ public static int GetLongestIonSeriesBidirectional(Dictionary<PeptideWithSetModi
             return maxDiffs.Max();
         }
 
+        /// <summary>
+        /// Determine the Fragment Coverage the PSM
+        /// Assigns fragment coverage indices for the PSM and the protein based on Amino Acid Position in Matched Ion Fragments
+        /// </summary>
+        public void GetAminoAcidCoverage()
+        {
+            if (this.BaseSequence != null && this.MatchedFragmentIons != null && this.MatchedFragmentIons.Any() &&
+                this.PeptideLength > 0)
```

Comment:
I think this can be shortened. suggest:
![image](https://user-images.githubusercontent.com/16841846/196790024-eb217e78-5bd5-4a4d-8377-54d40e877dee.png)

Reply:
I took out the check for peptide length, that one isn't necessary. I was still getting errors if I didn't have the MatchedFragmentIons null check, because it apparently can't do .Any() if the value is null. I also inverted it to reduce nesting.

Reply:
Updated some tests that created PSMs with null matched ion fragments, so now we don't have to check if the matchedFragmentIons is null.